tests/integration/conftest.py

```
@@ -0,0 +1,222 @@
+    crashed test or a SIGINT mid-suite would otherwise leak containers on the
+    runner. Returns count of reaped containers for visibility.
+    """
+    label = f"test-run-id={test_run_id}"
+    listing = subprocess.run(
+        ["docker", "ps", "-aq", "--filter", f"label={label}"],
+        capture_output=True, text=True,
+    )
+    ids = [i for i in listing.stdout.strip().splitlines() if i]
+    if not ids:
+        return 0
+    subprocess.run(["docker", "rm", "-f", *ids], capture_output=True)
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)